src/Tools/dotnet-monitor/Commands/ConfigShowCommandHandler.cs

```diff
@@ -20,8 +20,14 @@ public static void Invoke(string[] urls, string[] metricUrls, bool metrics, stri
             Write(Console.OpenStandardOutput(), urls, metricUrls, metrics, diagnosticPort, noAuth, tempApiKey, level);
         }
 
-        public static void Write(Stream stream, string[] urls, string[] metricUrls, bool metrics, string diagnosticPort, bool noAuth, bool tempApiKey, ConfigDisplayLevel level)
+        public static void Write(Stream stream, string[] urls, string[] metricUrls, bool metrics, string diagnosticPort, bool noAuth, bool tempApiKey, ConfigDisplayLevel level, string userSettingsFilePath = null, ConfigurationTestingMode testingMode = ConfigurationTestingMode.None)
         {
+            if (!string.IsNullOrEmpty(userSettingsFilePath))
+            {
+                HostBuilderHelper.UserSettingsPath = userSettingsFilePath;
```

Comment:
Statics will lead to problems, especially when two or more tests run at the same time. Or if one test sets them, but another test expects them to not be set or have their default values. When any of these occur, its really difficult to attempt to reproduce the problem.

Reply:
I was previously trying to use environment variables, but it looked like when I ran the tests only one would pass (because all the config show tests were sharing a single value for `userSettingsFilePath` instead of each using their own).

Reply:
The #1372 PR refactors host building so that all of the configuration directories can be specified via a settings. This allows dotnet-monitor itself to use environment variables and fixed paths for its configuration directories and allows tests to redirect those directories as they need. The PR allow adds some extension methods to override the configuration sources that provide configuration via environment variables to be replaced with in-memory sources for testing.